pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -0,0 +1,361 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package hpa
+
+import (
+	"k8s.io/api/autoscaling/v2beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"encoding/json"
+	"reflect"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	as "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+//// Covered by test/integration/util/kube_apiserver/hpa_test.go
+
+var (
+	expectedHPAType = reflect.TypeOf(v2beta1.HorizontalPodAutoscaler{})
+)
+
+const (
+	datadogHPAConfigMap = "datadog-hpa"
+)
+
+type CustomExternalMetric struct {
+	Name      string            `json:"name"`
+	Labels    map[string]string `json:"labels"`
+	Timestamp int64             `json:"ts"`
+	HpaOrigin string            `json:"origin"`
+	Value     int64             `json:"value"`
+}
+
+// HPAWatcherClient embeds the API Server client and the configuration to refresh metrics from Datadog and watch the HPA Objects' activities
+type HPAWatcherClient struct {
+	apiClient         *as.APIClient
+	hpaReadTimeout    time.Duration
```

Comment:
I think these field names are clear enough without the `hpa` prefix in the names.